supportedBackends/htcondor/src/main/scala/cromwell/backend/impl/htcondor/HtCondorWrapper.scala

```diff
@@ -99,23 +101,17 @@ class HtCondorProcess extends StrictLogging {
     */
   def jobReturnCode(jobId: String, returnCodeFilePath: Path): Int = {
 
-    @tailrec
-    def recursiveWait(): Int =
-      checkStatus(jobId) match {
-        case status if JobStatus.isTerminal(status) =>
-          Files.exists(returnCodeFilePath) match {
-            case true => File(returnCodeFilePath).contentAsString.stripLineEnd.toInt
-            case false =>
-              val msg = s"JobStatus from Condor is terminal ($status) and no RC file exists!"
-              logger.debug(msg)
-              throw new IllegalStateException(msg)
-          }
-        case nonTerminalStatus =>
-          Thread.sleep(5000)
-          recursiveWait()
-      }
-
-    recursiveWait()
+    checkStatus(jobId) match {
+      case status if JobStatus.isTerminal(status) =>
+        Files.exists(returnCodeFilePath) match {
+          case true => File(returnCodeFilePath).contentAsString.stripLineEnd.toInt
+          case false =>
+            val msg = s"JobStatus from Condor is terminal ($status) and no RC file exists!"
+            logger.debug(msg)
+            throw new IllegalStateException(msg)
+        }
+      case nonTerminalStatus => IncompleteRcIdentifier
```

Comment:
Why not just make the return type of the function an `Option[Int]` so that you can return `None` here and handle this case more officially where it gets called from